File: JAVASCRIPT/Cap06/c06ex03.js
// c06ex03.js

const rls = require('readline-sync');

class Aluno {
  constructor() {
    this.Nome = '';
    this.Turma = '';
    this.Sala = 0;
    this.Notas = new Array(4);
  }
}

let ALUNO = new Array(8).fill().map(() => new Aluno());

console.log("Escola de Computacao XPTO");
console.log("Cadastro Escolar de Aluno");
console.log("");

for (let I = 0; I <= 7; ++I) {
  console.log(`Aluno: ${I + 1}`);
  console.log("");

  ALUNO[I].Nome = rls.question("Nome ..............: ");
  ALUNO[I].Turma = rls.question("Turma .............: ");
  ALUNO[I].Sala = rls.questionInt("Sala ..............: ");

  console.log("");

  for (let J = 0; J <= 3; ++J)
    ALUNO[I].Notas[J] = rls.questionFloat(`Entre a ${J + 1}a. nota ..: `);

  console.log("");
}

console.log("Escola de Computacao XPTO");
console.log("Cadastro Escolar de Aluno");
console.log("");
console.log("DADOS DOS ALUNOS");
console.log("");

process.stdout.write("Aluno ");
process.stdout.write("Nome                           ");
process.stdout.write("Sala ");
process.stdout.write("Turma ");
process.stdout.write("Nota1 ");
process.stdout.write("Nota2 ");
process.stdout.write("Nota3 ");
console.log("Nota4");

process.stdout.write("----- ");
process.stdout.write("------------------------------ ");
process.stdout.write("---- ");
process.stdout.write("----- ");
process.stdout.write("----- ");
process.stdout.write("----- ");
process.stdout.write("----- ");
console.log("-----");

for (let I = 0; I <= 7; ++I) {
  process.stdout.write(`${(I + 1).toString().padStart(5, ' ')}`);
  process.stdout.write(` ${ALUNO[I].Nome.padEnd(30, ' ')}`);
  process.stdout.write(`${ALUNO[I].Sala.toString().padStart(5, ' ')}`);
  process.stdout.write(`${ALUNO[I].Turma.padStart(6, ' ')}`);
  for (let J = 0; J <= 3; ++J)
    process.stdout.write(`${ALUNO[I].Notas[J].toFixed(1).padStart(6, ' ')}`);
  console.log("");
}

console.log("");
process.stdout.write("Tecle <Enter> para encerrar... ");
rls.question();

File: JULIA/Cap06/c06ex03.jl
# c06ex03.jl

using Printf 

struct Aluno
  Nome::String
  Turma::Char
  Sala::Int
  Notas::Vector{Float64}
end

alunos = Array{Aluno, 1}(undef, 8)

println("Escola de Computacao XPTO")
println("Cadastro Escolar de Aluno")
println()

for i in 1:8
  println("Aluno: $i")
  println()

  print("Nome ..............: ")
  nome = readline()

  print("Turma .............: ")
  turma = first(readline())

  print("Sala ..............: ")
  sala = parse(Int, readline())
  println()

  notas = zeros(4)
  for j in 1:4
    print("Entre a $j", "a nota ...: ")
    notas[j] = parse(Float64, readline())
  end

  alunos[i] = Aluno(nome, turma, sala, notas)
  println()
end

println("Escola de Computacao XPTO")
println("Cadastro Escolar de Aluno")
println()
println("DADOS DOS ALUNOS")
print("Aluno ") 
print("Nome                           ")
print("Sala ")
print("Turma ")
print("Nota1 ")
print("Nota2 ")
print("Nota3 ")
println("Nota4 ")
print("----- ")
print("------------------------------ ")
print("---- ")
print("----- ")
print("----- ")
print("----- ")
print("----- ")
println("----- ")

for i in 1:8
  @printf("%5d ", i)
  @printf("%-30s ", alunos[i].Nome)
  @printf("%4d ", alunos[i].Sala)
  @printf("%5c ", alunos[i].Turma)
  for j in 1:4
    @printf("%5.1f ", alunos[i].Notas[j])
  end
  println()
end

println()
print("Tecle <Enter> para encerrar... ")
readline()
